packages/core/src/graph/graph.ts

```diff
@@ -1783,14 +1785,52 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
       comboConfig = combo;
     }
 
-    // step2: 更新 children，根据类型添加 comboId 或 parentId
+    // step 2: Pull children out of their parents
+    let comboTrees = this.get('comboTrees');
+    const isChildById = new Set(children);
```

Comment:
`childrenIds` is no good, because it implies a list though

Reply:
> On a related note, since we don't modify this variable, could we not just use children.includes instead of creating a dedicated set?

Using a list means you have to iterate through it if you want to find if an item belongs to the list. That's an `O(n)` operation, versus an `O(1)` operation you get from a dictionary. So no, dictionary is the right choice...

Reply:
Hmm... no, I'll keep using `Set` and `Map`. I'd have to change my code to:
```
    const childrenIdsSet = Object.create(null) as Record<string, boolean>;
    const pulledComboTreesById = Object.create(null) as Record<string, ComboTree>;
```

which is not ideal. I need to use `Object.create(null)` because I don't want keys inherited from `Object`'s prototype to be mistaken for valid keys (i.e. `pulledComboTreesById[id]` must not be unduly truthy)

Also, I'd have to initialize `childrenIdsSet` with `true` values for each element in the dictionary, which is a pain since we don't have `_.keyBy`...

No, let's keep using `Set` and `Map`. They are allowed, as shown in https://github.com/sirensolutions/G6/blob/master/packages/pc/src/util/combo-util.ts#L45-L46